evap/staff/tests.py

```diff
@@ -1081,28 +1085,96 @@ def load_test_data(self):
-        user2 = mommy.make(UserProfile, username="user2", course_set=[course2], delegates=[delegate2])
-        course2.voters = [user2]
-
-        self.assertEqual(UserProfile.objects.count(), 4)
-        call_command("merge_users", str(user1.pk), str(user2.pk), stdout=StringIO())
-        self.assertEqual(UserProfile.objects.count(), 3)
-
-        self.assertEqual(user1.pk, 3)
-        self.assertEqual(user1.username, "user1")
-        self.assertEqual(set(user1.course_set.all()), set([course1, course2]))
+    @classmethod
+    def setUpTestData(cls):
+        cls.user1 = mommy.make(UserProfile, username="test1")
+        cls.user2 = mommy.make(UserProfile, username="test2")
+        cls.user3 = mommy.make(UserProfile, username="test3")
+        cls.group1 = mommy.make(Group, name="group1")
+        cls.group2 = mommy.make(Group, name="group2")
+        cls.main_user = mommy.make(UserProfile,
+            username="main_user",
+            title="Dr.",
+            first_name="Main",
+            last_name="",
+            email="",  # test that merging works when taking the email from other user (UniqueConstraint)
+            groups=[cls.group1],
+            delegates=[cls.user1, cls.user2],
+            represented_users=[cls.user3],
+            cc_users=[cls.user1],
+            ccing_users=[]
+        )
+        cls.other_user = mommy.make(UserProfile,
+            username="other_user",
+            title="",
+            first_name="Other",
+            last_name="User",
+            email="other@test.com",
+            groups=[cls.group2],
+            delegates=[cls.user3],
+            represented_users=[cls.user1],
+            cc_users=[],
+            ccing_users=[cls.user1, cls.user2]
+        )
+        cls.course1 = mommy.make(Course, name="course1", participants=[cls.main_user, cls.other_user])  # this should make the merge fail
+        cls.course2 = mommy.make(Course, name="course2", participants=[cls.main_user], voters=[cls.main_user])
+        cls.course3 = mommy.make(Course, name="course3", participants=[cls.other_user], voters=[cls.other_user])
+        cls.contribution1 = mommy.make(Contribution, contributor=cls.main_user, course=cls.course1)
+        cls.contribution2 = mommy.make(Contribution, contributor=cls.other_user, course=cls.course1)  # this should make the merge fail
+        cls.contribution3 = mommy.make(Contribution, contributor=cls.other_user, course=cls.course2)
 
-        # see https://github.com/fsr-itse/EvaP/issues/705
-        #self.assertEqual(set(user1.delegates.all()), set([delegate1, delegate2]))
-        #self.assertEqual(set(course2.voters.all()), set([user1]))
+    def test_merge_users(self):
+        merged_user, errors = merge_users(self.main_user, self.other_user)  # merge should fail
+        self.assertSequenceEqual(errors, ['contributions', 'courses_participating_in'])
+
+        # assert that nothing has changed
+        self.assertEqual(self.main_user.username, "main_user")
```

Comment:
you will need to do a `self.main_user.refresh_from_db()`, otherwise the attributes are guaranteed to not change :)